src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp

```diff
@@ -0,0 +1,601 @@
+};
+
+constexpr Position RequiemOfTheAbyssPosition = { 3932.82f, -1243.81f, 128.368f };
+constexpr Position EruptingWatersSummonPosition1 = { 3917.31f, -1234.08f, 128.130f };
+constexpr Position EruptingWatersSummonPosition2 = { 3924.02f, -1259.52f, 128.141f };
+constexpr Position EruptingWatersSummonPosition3 = { 3950.80f, -1238.47f, 127.831f };
+
+struct AqusirrSharedAI : public BossAI
+{
+    AqusirrSharedAI(Creature* creature, uint32 bossId) : BossAI(creature, bossId)
+    {
+        SetBoundary(instance->GetBossBoundary(DATA_AQUSIRR));
+    }
+
+    virtual void ScheduleEvents()
+    {
+        events.ScheduleEvent(EVENT_SURGING_RUSH, 17s);
+        events.ScheduleEvent(EVENT_CHOKING_BRINE, 8900ms);
+        events.ScheduleEvent(EVENT_UNDERTOW, 36800ms);
+        events.ScheduleEvent(EVENT_SEA_BLAST, 1s);
+
+        if (IsHeroicOrHigher())
+            events.ScheduleEvent(EVENT_GRASP_FROM_THE_DEPTHS, 14300ms);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+        ScheduleEvents();
+    }
+
+    void ExecuteEvent(uint32 eventId) override
+    {
+        switch (eventId)
+        {
+            case EVENT_SEA_BLAST:
+            {
+                if (!me->GetCurrentSpell(CURRENT_CHANNELED_SPELL) && !me->IsWithinMeleeRange(me->GetVictim()))
+                {
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random))
+                        DoCast(target, SPELL_SEA_BLAST);
+                    events.Repeat(2400ms);
+                }
+                events.Repeat(1s);
+                break;
+            }
+            case EVENT_CHOKING_BRINE:
+            {
+                if (me->GetMap()->GetPlayersCountExceptGMs() > 1) // This event doesn't happen if you go alone
+                {
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random))
+                        DoCast(target, SPELL_CHOKING_BRINE);
+                }
+                events.Repeat(34s, 39s);
+                break;
+            }
+            case EVENT_SURGING_RUSH:
+            {
+                if (Unit* target = SelectTarget(SelectTargetMethod::Random))
+                    DoCast(target, SPELL_SURGING_RUSH);
+                events.RescheduleEvent(EVENT_SEA_BLAST, 10s);
+                events.Repeat(31500ms, 39300ms);
+                break;
+            }
+            case EVENT_UNDERTOW:
+            {
+                if (Unit* target = SelectTarget(SelectTargetMethod::Random))
+                    DoCast(target, SPELL_UNDERTOW);
+                events.Repeat(32400ms, 34s);
+                break;
+            }
+            case EVENT_GRASP_FROM_THE_DEPTHS:
+            {
+                DoCast(SPELL_GRASP_FROM_THE_DEPTHS_SELECTOR);
+                events.Repeat(38s);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
```

Comment:
you can drop UpdateAI entirely here, this behavior is default and ExecuteEvent is also called by default as you did it here)